compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -766,7 +767,9 @@ public BSymbol lookupConstructorSpaceSymbolInPackage(Location pos,
     }
 
     public BSymbol lookupMethodInModule(BPackageSymbol moduleSymbol, Name name, SymbolEnv env) {
-
+        if (moduleSymbol == null) {
```

Comment:
This is called by the programmer. It can happen if the programmer called for a module which is not loaded yet to memory.
Check the usage pattern around here.
https://github.com/ballerina-platform/ballerina-lang/blob/0418e82129063c92a2209aaf81d43ffde13bf62d/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java#L646

Do you think it is better to let it crash for NPE for an invalid call? Then yes this is not needed.

Reply:
Given that this wouldn't be null for user code, IMO it will be better if the developer can identify early on that the expected module is not available when we get here. WDYT?